controllers/topology/cluster_controller.go

```diff
@@ -100,13 +100,17 @@ func (r *ClusterReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manag
 		Controller: c,
 	}
 	r.patchEngine = patches.NewEngine()
```

Comment:
Should we maybe provide an alternate `SetupForDryRun` func which sets/initializes the patchEngine? This way we could also pass in a recorder and we don't have to expose the `Recorder` field.

Setting it in `Reconcile` when we only have to do it once instead of doing it during object construction feels hacky to me, especially as we set everything else in `SetupWithManager` (which is kind of a constructor for ClusterReconciler)